src/erlcloud_config.erl

```diff
@@ -0,0 +1,632 @@
+get_resource_config_history(Opts) ->
+    get_resource_config_history(Opts, default_config()).
+
+-spec get_resource_config_history(Opts :: map(),
+    Config :: aws_config()) ->
+    result_paged(map()).
+get_resource_config_history(Opts, Config) when is_map(Opts) orelse Opts =:= #{} ->
+    Request = #{?CHRONOLOGICAL_ORDER => maps:get(?CHRONOLOGICAL_ORDER, Opts, null),
+        ?EARLIER_TIME                => maps:get(?EARLIER_TIME, Opts, null),
+        ?LATER_TIME                  => maps:get(?LATER_TIME, Opts, null),
+        ?LIMIT                       => maps:get(?LIMIT, Opts, ?LIMIT_MAX),
+        ?NEXT_TOKEN_LABEL            => maps:get(?NEXT_TOKEN_LABEL, Opts, <<"">>),
+        ?RESOURCE_ID                 => maps:get(?RESOURCE_ID, Opts, null),
+        ?RESOURCE_TYPE               => maps:get(?RESOURCE_TYPE, Opts, null)},
+    case request(Config, "GetResourceConfigHistory", Request) of
+        {ok, Res} ->
+            {ok, Res, get_next_token(Res)};
+        {error,  _} = E -> E
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% ListDiscoveredResources
+%%
+%% https://docs.aws.amazon.com/config/latest/APIReference/API_ListDiscoveredResources.html
+%%------------------------------------------------------------------------------
+-spec list_discovered_resources() ->
+    result_paged(map()).
+list_discovered_resources() ->
+    list_discovered_resources(#{}).
+
+-spec list_discovered_resources(Opts :: map()) ->
+    result_paged(map()).
+list_discovered_resources(Opts) ->
+    list_discovered_resources(Opts, default_config()).
+
+-spec list_discovered_resources(Opts :: map(),
+    Config :: aws_config()) ->
+    result_paged(map()).
+list_discovered_resources(Opts, Config) when is_map(Opts) orelse Opts =:= #{} ->
+    Request = #{?INCLUDE_DELETED_RESOURCES => maps:get(?INCLUDE_DELETED_RESOURCES, Opts, true),
+        ?LIMIT                             => maps:get(?LIMIT, Opts, ?LIMIT_MAX),
+        ?NEXT_TOKEN_LABEL                  => maps:get(?NEXT_TOKEN_LABEL, Opts, <<"">>),
+        ?RESOURCE_IDS                      => maps:get(?RESOURCE_IDS, Opts, []),
+        ?RESOURCE_NAME                     => maps:get(?RESOURCE_NAME, Opts, null),
+        ?RESOURCE_TYPE                     => maps:get(?RESOURCE_TYPE, Opts, null)},
+    case request(Config, "ListDiscoveredResources", Request) of
+        {ok, Res} ->
+            {ok, Res, get_next_token(Res)};
+        {error,  _} = E -> E
+    end.
+
+%%%------------------------------------------------------------------------------
+%%% @doc
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+
+default_config() -> erlcloud_aws:default_config().
+
+request(Config0, OperationName, Request) ->
+    case erlcloud_aws:update_config(Config0) of
+        {ok, Config} ->
+            Body       = jsx:encode(Request),
+            Operation  = "StarlingDoveService." ++ OperationName,
```

Comment:
where does `StarlingDoveService` come from? I failed to find anything related on AWS website.

Reply:
@kkuzmin one of the ways is to see it turning debug mode for `aws cli`:

```
$ aws configservice describe-config-rules --profile mine --region us-east-1 --debug
2018-02-15 09:57:38,827 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.11.180 Python/3.6.3 Darwin/17.3.0 botocore/1.7.38
2018-02-15 09:57:38,827 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['configservice', 'describe-config-rules', '--profile', 'mine', '--region', 'us-east-1', '--debug']

...


2018-02-15 09:57:39,134 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=DescribeConfigRules) (verify_ssl=True) with params: {'url_path': '/', 'query_string': '', 'method': 'POST', 'headers': {'X-Amz-Target': 'StarlingDoveService.DescribeConfigRules', 'Content-Type': 'application/x-amz-json-1.1', 'User-Agent': 'aws-cli/1.11.180 Python/3.6.3 Darwin/17.3.0 botocore/1.7.38'}, 'body': b'{}', 'url': 'https://config.us-east-1.amazonaws.com/', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x105c435c0>, 'has_streaming_input': False, 'auth_type': None}}

...

content-type:application/x-amz-json-1.1
host:config.us-east-1.amazonaws.com
x-amz-date:20180215T095739Z
x-amz-target:StarlingDoveService.DescribeConfigRules

content-type;host;x-amz-date;x-amz-target

...


{
    "ConfigRules": []
}
```

Reply:
That came from hard graft.  I couldn't get this to work no matter what I tried.  Even the AWS console does not use the API directly, so it's hard to reverse engineer, and no doc tells you explicitly what to use.  Eventually I just got the AWS SDK and used Wireshark to see what it was using.  And presto!  It was "StarlingDoveService" and that worked. Weird, I know.  But what's a chap to do?